<?xml version="1.0"?>

<launch>

    <!-- Run the armor_service node -->
    <node name = "armor_service" 
          pkg="armor" 
          type = "execute" 
          args="it.emarolab.armor.ARMORMainService"/>


    <!-- Run the architecture's component. -->
    <rosparam param="state/initial_pose"> [0.0,  0.0] </rosparam>
    <rosparam param="config/environment_size"> [10.0, 10.0] </rosparam>

    <rosparam param="test/random_plan_points"> [2, 8] </rosparam>
    <rosparam param="test/random_plan_time"> [0.2, 0.8] </rosparam>
    <rosparam param="test/random_motion_time"> [0.1, 1.0] </rosparam>
    <rosparam param="test/random_sense/active"> True </rosparam>
    <rosparam param="test/random_sense/battery_time"> [15.0, 40.0] </rosparam>
    <!-- Run FSM and other nodes required -->
    <node pkg = "assignment"  
          type = "fsm.py"      
          name = "Finite_state_machine" 
          launch-prefix = "xterm -fa 'Monospace' -fs 10 -e"
    > </node>

    <node pkg = "assignment"  
          type = "robot_states.py" 
          name = "robot_states"
          
    > </node>
    
    <node pkg = "assignment"  
          type = "planner.py"
          name = "planner"
          
    > </node>
    
    <node pkg = "assignment"  
          type = "controller.py"
          name = "controller"
          
    > </node>
</launch>
